department-of-veterans-affairs/caseflow
# Shalliburton/caseflow 441 daily docket scheduled error

Resolves [CASEFLOW-441](https://vajira.max.gov/browse/CASEFLOW-441)

### Description
Adds the ability for hearing coordinators to change the disposition to `scheduled_in_error` from the daily docket page

### Acceptance Criteria

Daily Docket workflow:

- [ ] The Disposition dropdown for a hearing on the Daily Docket page has a new disposition: "Scheduled in Error"
- [ ] When a user selects that disposition, a modal appears that gives the user the following options:
  - [ ] Reschedule immediately
  - [ ] Send to Schedule Veterans list
- [ ] Ability to edit/add to the hearing Notes field (same notes field that appears on Hearing Details and Daily Docket)
- [ ] If the user chooses to "Reschedule immediately" and clicks the "Submit" button, they are taken to the Schedule Veteran page to schedule the hearing
- [ ] If the user chooses to "Send to Schedule Veterans list" and clicks the "Submit" button, they remain on the Daily Docket page with a success alert at the top of the page
- [ ] The success alert body includes a link back to Case Details for that appeal
- [ ] After the new hearing disposition has been confirmed, the hearing no longer displays on that Daily Docket page
- [ ] The disposition change for the hearing is displayed in Case Timeline just as it is for "postponed" dispositions
- [ ] Send the following Google Analytics custom event when a user submits the disposition change from Case Details or the Daily:

Category | Action | Label | Value
-- | -- | -- | --
"Hearings" | "Assign disposition: scheduled in error" | "Reschedule immediately" or "Send to Schedule Veteran list" | number of days after hearing was created

### Testing Plan
1. Login as `BVATWARNER`
2. go to http://localhost:3000/hearings/schedule and pick a hearing day that has veterans scheduled
3. > The disposition dropdown has a new option for scheduled hearings: "Scheduled in Error"
When a user selects that option, a modal appears giving the user the following options: `Reschedule immediately` and `Send to Schedule Veterans list`
4. Select the scheduled in error option from the disposition dropdown and Choose the "Send to Schedule Veterans list" and click the "Submit" button 
  - you should remain on the daily docket page with a success alert at the top of the page
  - you should also see the hearing removed from the daily docket
5. If the current daily docket does not have another veteran, choose a different daily docket
6. Select the scheduled in error option from the disposition dropdown and Choose to `Reschedule Immediately` option this time
  - choose `video`
     - fill out the form (leave something blank and expect error messages)
     - hit `submit` 
     - repeat steps 5 and 6
   - choose `virtual`
     - fill out form (enter invalid emails and expect error messages)
     - hit `submit` 
     - repeat steps 5 and 6

### User Facing Changes

 BEFORE|AFTER
 ---|---
| |<img width="1655" alt="DailyDocket-ScheduledInError-dropdown" src="https://user-images.githubusercontent.com/61989526/108277886-fbb04680-712e-11eb-9052-ede7ccc4529f.png">
| |<img width="1675" alt="DailyDocket-ScheduledInError-modal" src="https://user-images.githubusercontent.com/61989526/108277901-ffdc6400-712e-11eb-8da3-c349b92ab739.png">



### Code Documentation Updates
